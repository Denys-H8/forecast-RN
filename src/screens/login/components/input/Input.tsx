import React from 'react';
import { View, TextInput, TextInputProps } from 'react-native';

import constants from 'src/constants';
import { styles } from './input.styles';
import Icon from 'react-native-vector-icons/MaterialIcons';

type IProps = {
  error: boolean;
  iconName: string;
} & TextInputProps;

const Input: React.FC<IProps> = ({ error, iconName, ...TIProps }) => {
  const errorStyle = error ? { color: '#F00' } : {};

  return (
    <View>
      <View style={styles.inputContainer}>
        <Icon name={iconName} size={40} color={constants.BG_COLOR} />
        <TextInput style={[styles.input, errorStyle]} {...TIProps} testID="Input.TextInput" />
        {error && <Icon name="error-outline" size={30} color="#F00" testID="Input.Icon" />}
      </View>
    </View>
  );
};

export default Input;
