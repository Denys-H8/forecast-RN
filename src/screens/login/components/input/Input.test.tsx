import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import Input from './Input';

const styleMock = {
  backgroundColor: 'green',
};

const newTextValue = 'Some text value';

describe('Input ', () => {
  test('render input without error', () => {
    const onChangeTextMock = jest.fn();

    const { toJSON, queryByTestId, getByTestId } = render(
      <Input error={false} iconName="lock" onChangeText={onChangeTextMock} style={styleMock} />,
    );

    fireEvent.changeText(getByTestId('Input.TextInput'), newTextValue);

    expect(toJSON()).toMatchSnapshot();
    expect(onChangeTextMock).toHaveBeenCalled();
    expect(onChangeTextMock).toHaveBeenCalledWith(newTextValue);
    expect(onChangeTextMock).not.toHaveBeenCalledWith('Some other value');
    expect(queryByTestId('Input.Icon')).toBeNull();
  });

  test('render input with error', () => {
    const onChangeTextMock = jest.fn();

    const { toJSON, getByTestId } = render(
      <Input error={true} iconName="lock" onChangeText={onChangeTextMock} style={styleMock} />,
    );

    fireEvent.changeText(getByTestId('Input.TextInput'), newTextValue);

    expect(toJSON()).toMatchSnapshot();
    expect(onChangeTextMock).toHaveBeenCalled();
    expect(onChangeTextMock).toHaveBeenCalledWith(newTextValue);
    expect(onChangeTextMock).not.toHaveBeenCalledWith('Some other value');
    expect(getByTestId('Input.Icon')).not.toBeNull();
  });
});
