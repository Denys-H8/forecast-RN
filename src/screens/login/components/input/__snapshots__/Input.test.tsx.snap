// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input  render input with error 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#8B8EA5",
        "borderRadius": 100,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 8,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#191C39",
            "fontSize": 40,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={
        [MockFunction] {
          "calls": Array [
            Array [
              "Some text value",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "green",
        }
      }
      testID="Input.TextInput"
      underlineColorAndroid="transparent"
    />
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#F00",
            "fontSize": 30,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
      testID="Input.Icon"
    >
      
    </Text>
  </View>
</View>
`;

exports[`Input  render input without error 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#8B8EA5",
        "borderRadius": 100,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 8,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#191C39",
            "fontSize": 40,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={
        [MockFunction] {
          "calls": Array [
            Array [
              "Some text value",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "green",
        }
      }
      testID="Input.TextInput"
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;
